services/QuillLMS/app/workers/populate_aggregated_evidence_activity_healths_worker.rb

```diff
@@ -10,8 +10,9 @@ def perform
     # execute this command to start primary key indices back from 1
     ActiveRecord::Base.connection.execute("TRUNCATE evidence_activity_healths RESTART IDENTITY CASCADE")
 
+    evidence_activities = Evidence::Activity.all.reject { |a| a.flag == Flags::ARCHIVED.to_sym }
```

Comment:
So for this, `flag` is not a variable on the activity model but a method. (Because we store this variable on the parent activity class, not the evidence activity class). 

In my experience, we use scopes in cases where you essentially do an ActiveRecord query on the records in question, filtering by fields like in the example solution you gave me. I'm not sure how to do a scope where I want to filter not by field, but by the result of a certain method (`flag`) on the record?